test/foo/lib/config/test_configloader.py

```diff
@@ -0,0 +1,55 @@
+from unittest.mock import Mock
```

Comment:
The directories should follow pytest rules on tests. Please change this or create a `pytest.ini` file.
See for example: https://docs.pytest.org/en/latest/example/pythoncollection.html

I would rather change the files structure.